src/pages/index.js

```diff
@@ -4,81 +4,191 @@ import PopupWithForm from "../components/PopupWithForm.js";
+  avatarSelector: ".profile__avatar",
+});
+
+// ✅ Fetch user profile data and set it in the UI
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name || "Unknown Name",
+      job: userData.about || "No job specified",
+      avatar: userData.avatar || "default-avatar.jpg",
+    });
+  })
+  .catch((err) => console.error("Error fetching user data:", err));
+
+// ✅ Enable form validation
 const formValidators = {};
 const enableValidation = (config) => {
   const forms = Array.from(document.querySelectorAll(config.formSelector));
   forms.forEach((form) => {
     const validator = new FormValidator(config, form);
     const formName = form.getAttribute("name");
-    formValidators[formName] = validator; // Link validator to the form by its name
+    formValidators[formName] = validator;
     validator.enableValidation();
   });
 };
 enableValidation(settings);
 
-// Create a card
+// ✅ Create a card function
 function createCard(item) {
-  const card = new Card(item, ".card-template", handleImageClick);
-  return card.generateCard();
+  if (!item._id) {
+    console.error("Skipping card without ID:", item);
+    return null;
+  }
+  return new Card(
+    item,
+    ".card-template",
+    handleImageClick,
+    openDeletePopup,
+    api
+  ).generateCard();
 }
 
-// Initialize card section
+// ✅ Initialize card section (Only API Cards)
 const handleImageClick = (data) => imagePopup.open(data);
 const cardSection = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
+      if (cardElement) cardSection.addItem(cardElement);
     },
   },
   ".cards__list"
 );
-cardSection.renderItems();
 
-// Popup for image preview
+// ✅ Fetch and display only the current user's cards
+api
+  .getInitialCards()
+  .then((cards) => {
+    if (!Array.isArray(cards)) {
+      throw new Error("Invalid API response: Expected an array of cards");
+    }
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => console.error("Error loading initial cards:", err));
+
+// ✅ Popup for image preview
 const imagePopup = new PopupWithImage(".modal_type_image");
 imagePopup.setEventListeners();
 
-// Profile popup
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  jobSelector: ".profile__description",
-});
-
+// ✅ Profile popup
 const profilePopup = new PopupWithForm(
   ".modal_type_edit-profile",
   (data) => {
-    userInfo.setUserInfo({
-      name: data.name,
-      job: data.description,
-    });
+    return api
+      .updateUserInfo({
+        name: data.name,
+        about: data.description,
+      })
+      .then((updatedUser) => {
+        userInfo.setUserInfo({
+          name: updatedUser.name,
+          job: updatedUser.about,
+        });
+        document.forms["profile-form"].reset();
+      })
+      .catch((err) => console.error("Error updating user info:", err));
```

Comment:
You don't need `catch` in `index.js` if you have it inside `PopupWithForm` at the very end of the promise chain

**Please, correct it everywhere in the code**